modules/arcgis/src/deck-layer-view-2d.ts

```diff
@@ -1,26 +1,33 @@
-import {WebGLDevice} from '@luma.gl/webgl';
 import {initializeResources, render, finalizeResources} from './commons';
 
 export default function createDeckLayerView2D(BaseLayerViewGL2D) {
   return BaseLayerViewGL2D.createSubclass({
     properties: {
+      cancelInitialization: null,
       resources: null
     },
 
     // Attach is called as soon as the layer view is ready to start rendering.
-    attach() {
+    async attach() {
       // We use a full-screen quad and shaders to composite the frame rendered
       // with deck.gl on top of the MapView. Composition uses the MapView context.
       const gl = this.context;
-      const device = WebGLDevice.attach(gl);
 
-      this.resources = initializeResources.call(this, device);
+      let cancelled = false;
+      this.cancelInitialization = () => (cancelled = true);
+      const resources = await initializeResources.call(this, gl);
```

Comment:
Why `.call()`? Why do we need to set `this` rather than pass it as a param? Maybe a comment?

Reply:
Not introduced by this PR